### PR TITLE
Fix linting/formatting configuration

### PR DESCRIPTION
**Pull request by shonfeder**
Every time I try to work on a feature my editor gets confused with our linting configs. We also have inconsistent formatting across the code base, and we have missed some lints that can lead to errors (fixed in the last two commits here).

This PR aims to remove this friction by bringing our linting config in line with the documented guidance from eslint.

It also

- Adds a new CI job to enforce formatting/linting
- Adds an npm script as alias for formatting checks
- Fixes all the formatting variances and linting errors we've accrued

- [x] Tests added for any new code
- [ ] Ran `npm run format` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] Entries added to the respective `CHANGELOG.md` for any new functionality
- [ ] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

File: quint/src/IRprinting.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { QuintType, Row } from './quintTypes'
  * @returns a string with the pretty printed definition
  */
 export function moduleToString(quintModule: QuintModule): string {
-  const defs = quintModule.defs.map(d => definitionToString(d)).join('\n  ')
+  const defs = quintModule.defs.map((d) => definitionToString(d)).join('\n  ')
```
Comment (bugarela) [inline quint/src/IRprinting.ts:26]: Can we try to disable this rule? I dislike it.

File: quint/.prettierrc.js
Comment (bugarela) [inline quint/.prettierrc.js:1]: Why are we adding prettier? Is there something we can't accomplish with eslint?

**Review by bugarela — APPROVED** [review]

File: quint/.prettierrc.js
Comment (shonfeder) [reply quint/.prettierrc.js:1]: Sorry I forgot to ping this out, that is explained in the commit 

https://github.com/informalsystems/quint/pull/875/commits/e0ec926dc435c0a479035a06cd71ee7f764de8dc

other options are:

- don’t do any “formatting” fixes
- Ignore their recommendation

WDYT?

I don’t care either way tbh, so long as our configuration is consistent and enforced and doesn’t create friction when I’m trying to close features :)

File: quint/src/IRprinting.ts (hunk 1 above)
Comment (shonfeder) [reply quint/src/IRprinting.ts]: Disables in 65f662240706d57f5a967fbcfb08c47122addf91

Comment (bugarela) [reply quint/src/IRprinting.ts]: Thank you :tada:

File: quint/.prettierrc.js
Comment (bugarela) [reply quint/.prettierrc.js:1]: Looks like a good explanation to me, let's stick prettier than. I thought they were two tools for doing the same thing, but iiuc that's not the case.

**Pull request merged**